### PR TITLE
Remove toXContent implementations

### PR DESCRIPTION
**Pull request by mfussenegger**
See commits
Follow up to https://github.com/crate/crate/pull/16794 and https://github.com/crate/crate/pull/16832

There's more to remove - this is just a first pass

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/ElasticsearchException.java [hunk 1]
```diff
@@ -309,227 +287,6 @@ public static int getId(Class<? extends ElasticsearchException> exception) {
...
-    }
-
-    /**
-     * Generate a {@link ElasticsearchException} from a {@link XContentParser}. This does not
-     * return the original exception type (ie NodeClosedException for example) but just wraps
-     * the type, the reason and the cause of the exception. It also recursively parses the
-     * tree structure of the cause, returning it as a tree structure of {@link ElasticsearchException}
-     * instances.
-     */
-    public static ElasticsearchException fromXContent(XContentParser parser) throws IOException {
```
Comment (matriv) [inline server/src/main/java/org/elasticsearch/ElasticsearchException.java:398]: is this safe to remove? (bwc consideration).

File: server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java [hunk 2]
```diff
@@ -846,50 +843,50 @@ protected void ensureClusterSizeConsistency() {
...
-                    } catch (AssertionError error) {
-                        logger.error(
-                            "Cluster state from master:\n{}\nLocal cluster state:\n{}",
-                            masterClusterState.toString(),
-                            localClusterState.toString());
-                        throw error;
-                    }
-                }
-            }
+            // NamedWriteableRegistry namedWriteableRegistry = cluster().getNamedWriteableRegistry();
```
Comment (matriv) [inline server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java:846]: leftover?

File: server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java [hunk 3]
```diff
@@ -1092,16 +1089,16 @@ assert getLocalNode().equals(clusterState.nodes().get(getLocalNode().getId())) :
...
-            XContentHelper.convertToMap(
-                JsonXContent.JSON_XCONTENT, Strings.toString(event.previousState()), false
-            ).equals(
-                XContentHelper.convertToMap(
-                    JsonXContent.JSON_XCONTENT,
-                    Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState())),
-                    false))
-            : Strings.toString(event.previousState()) + " vs "
-            + Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState()));
+        // assert event.previousState() == coordinationState.get().getLastAcceptedState() ||
```
Comment (matriv) [inline server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java:1092]: leftover?

**Review by matriv — APPROVED** [review]
nice, left some comments though, please check.

File: server/src/main/java/org/elasticsearch/ElasticsearchException.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/ElasticsearchException.java:398]: Yes, it was only used by itself (recursively) and by `DefaultShardOperationFailedException.fromXContent` - which itself is also never used.

This was even unrelated to the storage changes and could've been removed before.

File: server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java]: Thanks for catching. Updated them.

**Pull request merged**